File: Singleton/python/singleton.py
class Car:

    __instance = None
    __private = False

    def __init__(self, color, capacity):
        if not Car.__private:
            print('this a singleton class')
        else:
            self.color = color
            self.capacity = capacity
            Car.__private = False
            Car.__instance = self

    @staticmethod
    def getInstance(color, capacity):
        if not Car.__instance:
            Car.__private = True
            Car(color, capacity)
        return Car.__instance


c = Car('sdsd', 1)
c1 = Car.getInstance('red', 12)
c = Car('sdsd', 1)
c2 = Car.getInstance('blue', 22)

print(id(c1))
print(id(c2))
print(c1.color)
print(c2.color)
print(c1 == c2)

File: Singleton/typeScript/singleton.ts
class Car {
  static _instance: Car;
  color: string;
  capacity: number;

  private constructor(color: string, capacity: number) {
    this.color = color;
    this.capacity = capacity;
  }

  static getInstance(color: string, capacity: number) {
      if(!this._instance){
        this._instance = new Car(color, capacity);
      }
      return this._instance;
  }
}

const c1 = Car.getInstance('red', 5)
const c2 = Car.getInstance('blue', 15)
console.log(c1)
console.log(c2)
console.log(c1==c2)